xwiki-rendering-transformations/xwiki-rendering-transformation-macro/src/main/java/org/xwiki/rendering/transformation/macro/MacroTransformationConfiguration.java

```diff
@@ -42,4 +47,14 @@
      *         descriptor (ie defined by the macro author) will be used
      */
     Properties getCategories();
+
+    /**
+     * @return the set of hidden categories (e.g., "Deprecated", "Internal")
+     * @since 14.8RC1
+     */
+    @Unstable
+    default Set<String> getHiddenCategories()
+    {
+        throw new UnsupportedOperationException();
```

Comment:
I would return an empty Set by default, otherwise there is no point in having a default implementation honestly if it's a failing one.

Reply:
I'm not sure since an empty set would be incorrect if there are hidden categories existing. I prefer the exception in this case as it's cleaner.

Reply:
The point f the default implementation is retro compatibility and since no category were hidden before that new API that's what the default behavior should reflect IMO.

Throwing an exception here make the default implementation totally useless, not implementing it at all would have the exact same effect. It's just lying to revapi basically.

Reply:
> Throwing an exception here make the default implementation totally useless

I don't agree with this statement. The point is to make any extension implementing the interface not fail with NoSuchMethodException when loaded by the CL. Any existing extension will not call the new method obviously since it's not been modified... So no backward compatibility issue here.

Reply:
(ofc if the extension is called by XWiki platform itself, then it needs to implement the new method or it'll fail)

Reply:
> The point is to make any extension implementing the interface not fail with NoSuchMethodException (not sure it's this exception but something related) when loaded by the CL.

There will be an exception only when the method is called, not when the class is loaded.

Reply:
Thanks for the link, it's well documented there. I was wrong as I thought the linker would check for missing method implementations during CL. So no backward compatibility issue when adding new methods to interfaces (good to know, I think I've been wrong since the beginning when I proposed to use default methods when moving to Java 8 :)). 

So this means that we don't really need to add a default impl when adding a new interface method for which we throw an UnsupportedOperationException exception. See the numerous places where we do that at https://github.com/xwiki/xwiki-platform/search?q=new+UnsupportedOperationException

Reply:
> So no backward compatibility issue when adding new methods to interfaces

I don't agree with that. It's clearly a retro compatibility when the interface might be implemented by an extension and used by another or XWiki Standard which knows about the new method, the default implementation should try to return something that make sense in such a case (here what make sense is to indicate that there is no hidden macros since that custom implementation of MacroTransformationConfiguration does not have any way to configure hiden macros).

> See the numerous places where we do that at [xwiki/xwiki-platform/search (new UnsupportedOperationException)](https://github.com/xwiki/xwiki-platform/search?q=new+UnsupportedOperationException)

Not all of them are about default implementation of new methods, but those that are are wrong in most cases IMO. Did not notice some people had done it. We should always try to find a not failing default behavior, focused on retro compatibility.

Reply:
> So no backward compatibility issue when adding new methods to interfaces

I meant no *binary* backward compatibility issue.

It's a problem when it's a SPI as mentioned at https://revapi.org/revapi-java/0.26.1/differences.html#java.method.addedToInterface